vms/avm/service.go

```diff
@@ -815,19 +789,55 @@ func (s *Service) buildCreateAssetTx(args *CreateAssetArgs) (*txs.Tx, ids.ShortI
 	codec := s.vm.parser.Codec()
 	initialState.Sort(codec)
 
-	tx := &txs.Tx{Unsigned: &txs.CreateAssetTx{
+	return buildCreateAssetTx(
+		s.vm,
+		args.Name,
+		args.Symbol,
+		args.Denomination,
+		[]*txs.InitialState{initialState},
+		utxos,
+		kc,
+		changeAddr,
+	)
+}
+
+func buildCreateAssetTx(
+	vm *VM,
+	name, symbol string,
+	denomination byte,
+	initialStates []*txs.InitialState,
+	utxos []*avax.UTXO,
+	kc *secp256k1fx.Keychain,
+	changeAddr ids.ShortID,
+) (*txs.Tx, ids.ShortID, error) {
+	uTx := &txs.CreateAssetTx{
 		BaseTx: txs.BaseTx{BaseTx: avax.BaseTx{
-			NetworkID:    s.vm.ctx.NetworkID,
-			BlockchainID: s.vm.ctx.ChainID,
-			Outs:         outs,
-			Ins:          ins,
+			NetworkID:    vm.ctx.NetworkID,
+			BlockchainID: vm.ctx.ChainID,
 		}},
-		Name:         args.Name,
-		Symbol:       args.Symbol,
-		Denomination: args.Denomination,
-		States:       []*txs.InitialState{initialState},
-	}}
-	return tx, changeAddr, tx.SignSECP256K1Fx(codec, keys)
+		Name:         name,
+		Symbol:       symbol,
+		Denomination: denomination,
+		States:       initialStates,
+	}
+
+	toBurn := map[ids.ID]uint64{
+		vm.feeAssetID: vm.CreateAssetTxFee,
+	}
+	ins, outs, keys, err := vm.NewSpend(
```

Comment:
NewSpend is temporary. I plan to remove the current Spend and replace it with NewSpend (which is much more similar to what we do in the wallet)